src/background/index.js

```diff
@@ -1,237 +1,305 @@
+    this._hasViewPort = false
 
-      chrome.browserAction.setIcon({ path: './images/icon-green.png' })
-      chrome.browserAction.setBadgeText({ text: this._badgeState })
-      chrome.browserAction.setBadgeBackgroundColor({ color: '#FF0000' })
+    await browser.injectContentScript()
+    this.toggleOverlay({ open: true, clear: true })
 
-      /**
-       * Right click menu setup
-       */
+    this._boundedMessageHandler = this.handleMessage.bind(this)
+    this._boundedNavigationHandler = this.handleNavigation.bind(this)
+    this._boundedWaitHandler = () => badge.wait()
 
-      chrome.contextMenus.removeAll()
+    this.overlayHandler = this.handleOverlayMessage.bind(this)
 
-      // add the parent and its children
+    // chrome.contextMenus.create({
+    //   id: this._menuId,
+    //   title: 'Headless Recorder',
+    //   contexts: ['all'],
+    // })
 
-      chrome.contextMenus.create({
-        id: this._menuId,
-        title: 'Headless Recorder',
-        contexts: ['all']
-      })
+    // chrome.contextMenus.create({
+    //   id: this._menuId + 'SELECTOR',
+    //   title: 'Copy Selector',
+    //   parentId: this._menuId,
+    //   contexts: ['all'],
+    // })
 
-      chrome.contextMenus.create({
-        id: this._menuId + this._menuOptions.SCREENSHOT,
-        title: 'Take Screenshot (Ctrl+Shift+A)',
-        parentId: this._menuId,
-        contexts: ['all']
-      })
+    // this._boundedMenuHandler = this.handleMenuInteraction.bind(this)
+    // chrome.contextMenus.onClicked.addListener(this._boundedMenuHandler)
 
-      chrome.contextMenus.create({
-        id: this._menuId + this._menuOptions.SCREENSHOT_CLIPPED,
-        title: 'Take Screenshot Clipped (Ctrl+Shift+S)',
-        parentId: this._menuId,
-        contexts: ['all']
-      })
+    chrome.runtime.onMessage.addListener(this._boundedMessageHandler)
+    chrome.runtime.onMessage.addListener(this.overlayHandler)
 
-      // add the handlers
+    chrome.webNavigation.onCompleted.addListener(this._boundedNavigationHandler)
+    chrome.webNavigation.onBeforeNavigate.addListener(this._boundedWaitHandler)
 
-      this._boundedMenuHandler = this.handleMenuInteraction.bind(this)
-      chrome.contextMenus.onClicked.addListener(this._boundedMenuHandler)
-
-      this._boundedKeyCommandHandler = this.handleKeyCommands.bind(this)
-      chrome.commands.onCommand.addListener(this._boundedKeyCommandHandler)
-    })
+    badge.start()
   }
 
-  stop () {
-    console.debug('stop recording')
+  stop() {
     this._badgeState = this._recording.length > 0 ? '1' : ''
 
     chrome.runtime.onMessage.removeListener(this._boundedMessageHandler)
     chrome.webNavigation.onCompleted.removeListener(this._boundedNavigationHandler)
     chrome.webNavigation.onBeforeNavigate.removeListener(this._boundedWaitHandler)
-    chrome.contextMenus.onClicked.removeListener(this._boundedMenuHandler)
+    // chrome.contextMenus.onClicked.removeListener(this._boundedMenuHandler)
```

Comment:
this should be commented out?

Reply:
Is a new feature I am working on but not finished yet